### PR TITLE
Feature #### signup bug fixes

### PR DESCRIPTION
**Pull request by Mosuhli-Coder**

### CODE REVIEW

File: controllers/authentication.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ exports.adminAccount = async (req, res) =>{
 
     }catch(error){
         // console.log(error)
-        return res.sendStatus(500)
+        return res.send("Please fill out all the fields")
```
Comment (eliasbayona) [inline controllers/authentication.js:46]: return appropriate server status code

File: controllers/authentication.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ exports.register = async (req, res) =>{
...
```
Comment (eliasbayona) [inline controllers/authentication.js:24]: return appropriate respond status code

File: routes/login.js [hunk 3]
```diff
@@ -29,21 +29,14 @@ router.post(
       const { username, password } = req.body;
       try {
         let user = await User.findOne({
-          username
+          "profile.name.screenName":username,
```
Comment (eliasbayona) [inline routes/login.js:32]: screen names are not unique, use email instead

File: routes/login.js [hunk 4]
```diff
@@ -29,21 +29,14 @@ router.post(
...
+           password: password
```
Comment (eliasbayona) [inline routes/login.js:33]: not sure why you moved the previous block of code, at the current moment you cannot compare a hashed password string with a plain text password string

**Review by eliasbayona — CHANGES_REQUESTED** [review]

File: routes/login.js (hunk 4 above)
Comment (Mosuhli-Coder) [reply routes/login.js:33]: The problem is admin is the one whose password is stored and the password its just plain, i don't know what i have to do, please help

File: controllers/authentication.js [hunk 5]
```diff
@@ -35,15 +35,16 @@ exports.adminAccount = async (req, res) =>{
           password,
           confirmPassword
         } = req.body
-
-
         let registeredUser = await authUtils.registerAdmin(password, confirmPassword, email, channel, username)
-        res.send(registeredUser)
+        if(password === confirmPassword){
```
Comment (eliasbayona) [inline controllers/authentication.js:39]: take another look at the flow of your logic here.

You seem to be creating an admin user before you actually check if the preconditions to create the user pass.
i.e you generally want to do all the password checks before creating the user.

**Review by eliasbayona — CHANGES_REQUESTED** [review]

File: controllers/authentication.js (hunk 5 above)
Comment (Mosuhli-Coder) [reply controllers/authentication.js:39]: Meaning i should check if the password matches, if not, i should not send data,  else send data, i don't understand

**Pull request merged**